Better room modification CLI arguments; allow --description modification

jagerman opened the pull request:
Room modification arguments were sort of a shambles: there were various "action" items like "--set-perms" or "--add-moderator", but those options were exclusive (meaning you couldn't, say, add a moderator and delete another moderator in the same command).

This overhauls how modifications work, accomplishing the following:
- removed the `--set-perms` argument; it is implied by having any of the `--add-perms`/`--remove-perms`/`--clear-perms` options.
- `--description` can now be used to update a room description (when used with `--rooms` to specify the room).  Previously it was only used when adding a room.
- Modifying options (like `--description`, `--add-moderators`, etc.) are no longer mutually exlusive: one command can now add/delete moderators, set permissions, and change the description all in one go.
- Simplified various identical logic around how `--rooms` is handled.

jagerman committed:
Better room modification CLI arguments; allow --description modification

Room modification arguments were sort of a shambles: there were various
"action" items like "--set-perms" or "--add-moderator", but those
options were exclusive (meaning you couldn't, say, add a moderator and
delete another moderator in the same command).

This overhauls how modifications work, accomplishing the following:
- removed the `--set-perms` argument; it is implied by having any of the
  `--add-perms`/`--remove-perms`/`--clear-perms` options.
- `--description` can now be used to update a room description (when
  used with `--rooms` to specify the room).  Previously it was only used
  when adding a room.
- Modifying options (like `--description`, `--add-moderators`, etc.) are
  no longer mutually exlusive: one command can now add/delete
  moderators, set permissions, and change the description all in one go.
- Simplified various identical logic around how `--rooms` is handled.
- Add '\n' and '\\' escapes to the `--description` argument so that you
  can add newlines into the description without having to fumble around
  with passing literal newlines in cli arguments.

tewinget commented on sogs/__main__.py:
```diff
@@ -147,6 +161,40 @@
+
+if update_room and not args.rooms:
+    print(
+        "A room must be specified (using --rooms) when updating permissions or room details",
```
```suggestion
        "A room must be specified (using --rooms) when updating room permissions or details",
```

tewinget replied:
nevermind, misunderstood

jagerman committed:
Improve errors

- the `--rooms` code/error message was potentially confusing
- some errors weren't going to stderr

jagerman committed:
Improve `--verbose` description
